packages/react-components/react-field/Spec.md

```diff
@@ -2,62 +2,341 @@
+- `SelectField`
+- `SliderField`
+- `SpinButtonField`
+- `SwitchField`
+  - _Open question:_ how should this handle the `label`? Should it forward to the underlying `Switch`, or keep the label in the same place as the field's label? Might need a prop to control this behavior.
+- `TextareaField`
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+- **Orientation**: The `orientation` prop affects the layout of the label and field component:
+  - `'vertical'` (default) - label is above the field component
+  - `'horizontal'` - label is to the left of the field component, and is 33% the width of the field (this allows multiple stacked fields to all align their labels)
+- **Validation state**: The `validationState` prop affects the icon and color used by the `validationMessage`:
+  - `'error'` - Red x icon, red text color
+  - `'warning'` - Yellow exclamation icon, neutral color text
+  - `'success'` - Green check icon, neutral color text
+  - `undefined` (default): No validation message icon, neutral color text
+- **Error**: Some control types (like `Input` and `Combobox`) have a prop that makes the border red. This prop will be set `validationState="error"`.
+
+Field also forwards some props from the wrapped component to the label as well:
+
+- **Size**: If the wrapped component supports a `size` prop, it will also be applied to the field's label.
+- **Required**: If set, the Label will get a required asterisk: `*`
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+### FieldComponent
+
+The `FieldComponent` type defines the minimum set of props that the wrapped component must support. This is used for the generic types as the requirement for the type parameter: `FieldProps<T extends FieldComponent>`
+
+```ts
+/**
+ * The minimum requirement for a component used by Field.
+ *
+ * Note: the use of VoidFunctionComponent means that component is not *required* to have a children prop,
+ * but it is still allowed to have a children prop.
+ */
+export type FieldComponent = React.VoidFunctionComponent<
+  Pick<
+    React.HTMLAttributes<HTMLElement>,
+    'id' | 'className' | 'style' | 'aria-labelledby' | 'aria-describedby' | 'aria-invalid' | 'aria-errormessage'
+  >
+>;
+```
+
+### Slots
+
+_Note: TypeScript crashes if the `Slot` type is used with a template type parameter. The `SlotComponent` type is a simplified version of that type, which only supports `React.ComponentType`/`React.VoidFunctionComponent`._
+
+```ts
+export type FieldSlots<T extends FieldComponent> = {
+  root: NonNullable<Slot<'div'>>;
+
+  /**
+   * The underlying component wrapped by this field.
+   *
+   * This is the PRIMARY slot: all intrinsic HTML properties will be applied to this slot,
+   * except `className` and `style`, which remain on the root slot.
+   */
+  field: SlotComponent<T>;
```

Comment:
FYI, I changed this from `fieldComponent` in earlier iterations, to just `field`. The old name seemed awkward, but the new name might be potentially confusing... I liken it to the `Input` component's `input` slot to access the underlying input element.

Does anyone have opinions about the name here?

Reply:
Agree with you, just `field` is better 👍 Another option to consider is `control` like in [SUIR](https://react.semantic-ui.com/collections/form/#shorthand-field-control-html)

Reply:
I like `control` -- I'll change it to that, thanks for the suggestion!